### ISSUE TITLE
Name conflict between 'filter' and 'where'

### ISSUE THREAD

**Issue by ani-sha** [graphqlcrud/graphqlcrud-java#43]
```
type Mutation {
  updateNote(input: MutateNoteInput!, where: UpdateNoteFilter): Note!
}
```
We have been using `filter` in GrapgQL CRUD Java and encountered this conflict, `where` and `filter` function the same.
`where` sounds more SQL specific, can be replaced with `filter` in GraphQL CRUD spec.

IMO, a `delete` should not require an `input` argument name, it should be `filter`. 
```
type Mutation {
    deleteNote(input: MutateNoteInput!): Note
}
```
Suggested type defination:
```
type Mutation {
    deleteNote(filter: UpdateNoteFilter!): Note
}
```
cc: @wtrocki @machi1990 @rareddy

**Comment by machi1990** [comment]
> IMO, a `delete` should not require an `input` argument name, it should be `filter`.
> 
> ```
> type Mutation {
>     deleteNote(input: MutateNoteInput!): Note
> }
> ```
> 
> Suggested type defination:
> 
> ```
> type Mutation {
>     deleteNote(filter: UpdateNoteFilter!): Note
> }
> ```
> 
> cc: @wtrocki @machi1990 @rareddy

+1. This makes total sense to me. 

/cc @craicoverflow for additional insights.

**Comment by craicoverflow** [comment]
Makes sense to me also, `input` makes more sense when inserting data. I guess it was used because it is a mutation.

It would be great if you could copy this issue to https://github.com/graphqlcrud/spec, where it can be discussed in a wider context as there are other contributors who may be interested in this discussion.

**Comment by machi1990** [comment]
> It would be great if you could copy this issue to https://github.com/graphqlcrud/spec, where it can be discussed in a wider context as there are other contributors who may be interested in this discussion.

+100.
